File: python/pe021.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 021 """

import sys
from math import sqrt

def sum_factors(number):
    total = 1
    for i in range(2, int(sqrt(number))):
        if ((number % i) == 0):
            total += (i + (number // i))

    return total

SIZE = 10000
spd = []
for i in range(SIZE):
    spd.append(sum_factors(i))

result = 0
for i in range(SIZE):
    if (spd[i] < SIZE):
        if (i != spd[i]):
            if (spd[spd[i]] == i):
                result += i

print(result)
sys.exit(0)

File: lua/pe021.lua
-- this computes projecteuler.net problem 021

local SIZE = 10000
local spd = {}

local function sum_factors(number)
  local total = 1
  for i = 2, math.floor(math.sqrt(number)) do
    if (number % i) == 0 then
      total = i + math.floor(number / i) + total
    end
  end
  return total
end

for i = 1, SIZE do
  spd[i] = sum_factors(i)
end

local result = 0
for i = 1, SIZE do
  if spd[i] < SIZE then
    if i ~= spd[i] then
      if spd[spd[i]] == i then
        result = i + result
      end
    end
  end
end

print(result)
